src/vs/workbench/contrib/relauncher/browser/relauncher.contribution.ts

```diff
@@ -258,6 +261,50 @@ export class WorkspaceChangeExtHostRelauncher extends Disposable implements IWor
 	}
 }
 
+export class RuntimeArgumentsChangeRelauncher extends Disposable implements IWorkbenchContribution {
+
+	private readonly disableChromiumSandbox = new ChangeObserver('boolean');
+
+	constructor(
+		@IHostService private readonly hostService: IHostService,
+		@IFileService private readonly fileService: IFileService,
+		@IEnvironmentService private readonly environmentService: IEnvironmentService,
+		@IProductService private readonly productService: IProductService,
+		@IDialogService private readonly dialogService: IDialogService
+	) {
+		super();
+
+		this._register(this.fileService.watch(this.environmentService.argvResource));
+		this._register(this.fileService.onDidRunOperation((e) => {
+			if (e.isOperation(FileOperation.WRITE) && e.resource.toString() === this.environmentService.argvResource.toString()) {
+				this.onRuntimeArgumentFileEdited();
+			}
+		}));
+	}
+
+	private async onRuntimeArgumentFileEdited(): Promise<void> {
+		const argvContent = await this.fileService.readFile(this.environmentService.argvResource);
+		const argvString = argvContent.value.toString();
+		const argvJSON = JSON.parse(stripComments(argvString));
+		// Chromium sandbox
+		const changed = this.disableChromiumSandbox.handleChange(argvJSON['disable-chromium-sandbox']);
+
+		// Notify only when changed from an event and the change
+		// was not triggerd programmatically (e.g. from experiments)
+		if (changed && this.hostService.hasFocus) {
```

Comment:
Why do we actually do this kind of check? Why not always restart when the file is written to, what about the other entries in the file, they all require a restart, no? I guess with the exception of the machine ID, which I think is set from the main process.

Reply:
Oh nevermind my idea to always restart on each write, that would be an ugly experience when the user is typing but has not finished typing. 

Anyway, we could still restart for when any of the keys change that we know are interesting?

Reply:
Yup I am good with checking the other keys, I wanted to get a green on the sandbox setting before adding more entries.